Tests/Runner/RunnerTest.php

```diff
@@ -799,6 +800,446 @@ public function testExecutorStoreStateWithProcessor()
+                    '   print("foo,bar\n1,2", file=out_table)',
+                    'with open("/data/out/tables/out.c-runner-test.my-table-2b.csv", "w") as out_table:',
+                    '   print("foo,bar\n1,2", file=out_table)',
+                ],
+            ],
+        ];
+        $configurationRow->setConfiguration($configData2);
+        $component->addConfigurationRow($configurationRow);
+
+        $componentData = [
+            'id' => 'keboola.docker-demo-sync',
+            'data' => [
+                'definition' => [
+                    'type' => 'aws-ecr',
+                    'uri' => '147946154733.dkr.ecr.us-east-1.amazonaws.com/developer-portal-v2/keboola.python-transformation',
+                ],
+            ],
+        ];
+        $runner = $this->getRunner();
+        try {
+            $runner->run(
+                [
+                    new JobDefinition($configData1, new Component($componentData), 'runner-configuration', 'v123', [], 'row-1'),
+                    new JobDefinition($configData2, new Component($componentData), 'runner-configuration', 'v123', [], 'row-2'),
+                ],
+                'run',
+                'run',
+                '1234567',
+                new NullUsageFile()
+            );
+        } catch (UserException $e) {
+            self::assertContains(
+                'Cannot upload file \'out.c-runner-test.my-table-1b.csv\' to table ' .
+                '\'out.c-runner-test.my-table-1b\' in Storage API: There are duplicate columns in CSV file: foo',
+                $e->getMessage()
+            );
+        }
+
+        $component = new Components($this->getClient());
+        $listOptions = new ListConfigurationRowsOptions();
+        $listOptions->setComponentId('keboola.docker-demo-sync')->setConfigurationId('runner-configuration');
+        $configuration = $component->getConfiguration('keboola.docker-demo-sync', 'runner-configuration');
+        // configuration state should be unchanged
+        self::assertArrayHasKey('foo', $configuration['state']);
+        self::assertEquals('bar', $configuration['state']['foo']);
+        $rows = $component->listConfigurationRows($listOptions);
+        uasort(
+            $rows,
+            function ($a, $b) {
+                return strcasecmp($a['id'], $b['id']);
+            }
+        );
+        $row1 = $rows[0];
+        $row2 = $rows[1];
+        self::assertArrayNotHasKey('bazRow1', $row1['state']);
+        self::assertArrayNotHasKey('bazRow2', $row2['state']);
+        self::assertFalse($this->client->tableExists('out.c-runner-test.my-table-1b'));
+        self::assertFalse($this->client->tableExists('out.c-runner-test.my-table-2a'));
+        self::assertFalse($this->client->tableExists('out.c-runner-test.my-table-2b'));
+        $this->clearConfigurations();
+    }
+
+    public function testExecutorStoreStateRowsOutputMappingDeferredEarlyError()
```

Comment:
a tohle je 1:1 stejný jak předchozí test, jen s featurou, ne? tak bych to udělal jako jeden test a předal tomu featuru přes dataProvider

Reply:
done